src/project/manifest/mod.rs

```diff
@@ -218,6 +225,60 @@ impl Manifest {
         Ok(())
     }
 
+    /// Adds an environment to the project. Overwrites the entry if it already
+    /// exists.
+    pub fn add_environment<'f>(
+        &mut self,
+        name: &str,
+        features: impl IntoIterator<Item = &'f FeatureName>,
+        solve_group: Option<&str>,
+        no_default_feature: bool,
+    ) -> miette::Result<()> {
+        let features = features.into_iter().cloned().collect_vec();
+
```

Comment:
```suggestion

        // Check if features exist
        for feature in &features {
            if self.feature(feature).is_none() {
                return Err(miette!(
                    "feature `{}` does not exist, existing features: {}",
                    feature.fancy_display(),
                    self.parsed
                        .features
                        .keys()
                        .map(|f| f.fancy_display())
                        .join(", ")
                ));
            }
        }

```
Would like to check it before we write it to file as it will leave the file in a broken state and then you can not use the cli to edit it anymore.

Reply:
I modified it even further: It now throws this error:
```
 × the feature 'henk' is not defined in the project manifest
  help: Did you mean 'schema'?
```